### ISSUE TITLE
Latest update not picking up styles.less

### ISSUE THREAD

**Issue by NV-GCK** [travs/markdown-pdf#165]
I am currently using Atom on Win 7 64 bit.  Atom version is 1.19.4 x64, and markdown-pdf version is 2.0.3. The generated PDF is completely ignoring my styles.less. It was working just fine before your most recent update.

**Comment by NV-GCK** [comment]
Just verified on my Mac at home as well. Running OS X 10.10.5; Atom version 1.19.5 x64. The previous version 1.5.4 that was on my Mac was perfectly picking up the markdown formatting in my styles.less. The current version 2.0.3 completely ignores my styles.less and seems to be formatting straight GFM.

**Comment by NV-GCK** [comment]
Please help, please and thank you.

**Comment by travs** [comment]
@NV-GCK 
Thanks for the report.
I will make a fix for this today. Will ping you again when I update.

**Comment by travs** [comment]
@NV-GCK Updated. Try it now and see if it works for you!

**Comment by NV-GCK** [comment]
No luck, tried uninstalling and re-installing. Still stuck at version 2.0.4, still not picking up styles.less, but thank you regardless for looking into this. Hope this gets sorted by 2.0.5.

**Comment by travs** [comment]
@NV-GCK Huh, ok I will re-open until this is fixed.
Can you tell me what you get if you open the atom dev console and do `atom.styles.getUserStyleSheetPath();`? This is the path of the `styles.less` you're talking about right?

**Comment by NV-GCK** [comment]
The path is actually pointing to my style sheet as expected, but it's still not picking up the specifics in my styles.less. Thanks and best.

**Comment by madsesmith** [comment]
@travs I'm having the same problem that @NV-GCK describes. Previously when my converted PDFs used the styles in my `styles.less` file and resulted in PDFs with a smaller font. Now I'm getting PDFs with large fonts again, like before I modified the `styles.less` file. My `markdown-preview` does still show the correct style. I also checked the path and it points to the stylesheet I've been working with. 

markdown-pdf 2.0.4
atom 1.19.3
macOS 10.12.6

**Comment by travs** [comment]
@madsesmith Thanks for confirming this bug. Will investigate this right now.

For the present moment, I have added an option to the config that force-enables "fallback mode", so you can get pdf output from the latest "stable" (`1.x.x`) version of the package.

There have been some breaking changes in `2.x.x` (as you've seen), but if you update to `2.1.0`, then it might still pick up your `styles.less` while I work towards a patch for the present issue.

**Comment by travs** [comment]
@madsesmith @NV-GCK 
Can either of you provide some sample selectors from your `styles` for me to try?
Perhaps some work and others do not, or something.

For instance, on both linux and windows7, this parameter definitely produces differing output for me:
```less
body, p, li {
  font-size: 2rem;
}
```

If either of you can dump your `styles.less` I can give the specific file a try locally as well; may help diagnose.

It may also be something to do with the way the package passes along the styles to the converter module: 

1. compiling the `.less` file contents to css
2. inserting this into a newly-created temporary `.css` file
3. passing the file handle to the module (e.g. `/tmp/sadn74d` on linux, but differs by OS).

**Comment by NV-GCK** [comment]
/*
 * Your Stylesheet
 *
 * This stylesheet is loaded when Atom starts up and is reloaded automatically
 * when it is changed and saved.
 *
 * Add your own CSS or Less to fully customize Atom.
 * If you are unfamiliar with Less, you can read more about it here:
 * http://lesscss.org
 */


/*
 * Examples
 * (To see them, uncomment and save)
 */

// style the background color of the tree view
.tree-view {
  // background-color: whitesmoke;
}

// style the background and foreground colors on the atom-text-editor-element itself
atom-text-editor {
  // color: white;
  // background-color: hsl(180, 24%, 12%);
}

// style UI elements inside atom-text-editor
atom-text-editor .cursor {
  border-color: blueviolet;
}

.markdown-preview.markdown-preview {
  p, ul {
  font-family: "Times New Roman";
  font-size: 12px;
  }

  h1{
  font-family: "Palatino Linotype", Palatino, serif;
  font-size: 20px;
  }

  h2{
  font-family: "Palatino Linotype", Palatino, serif;
  font-size: 18px;
  }

  h3{
  font-family: "Palatino Linotype", Palatino, serif;
  font-size: 16px;
  }

  h4{
  font-family: "Helvetica-Neue", Helvetica, sans-serif;
  font-size: 14px;
  }

  h5 {
  font-family: "Helvetica-Neue", Helvetica, sans-serif;
  font-size: 12px;
  }

  h6 {
  font-family: "Helvetica-Neue", Helvetica, sans-serif;
  font-size: 10px;
  }
}

**Comment by NV-GCK** [comment]
Version 2.1.0 not working in fallback or non-fallback mode. Verified that Atom is picking up my stylesheet via the atom.styles.getUserStyleSheetPath(); command in the console.

**Comment by madsesmith** [comment]
Additions to my `styles.less`:

```
.markdown-preview[data-use-github-style]{
  font-size: 11px;
  
  h1 {
    font-size: 1.3em;
  }
  h2 {
    font-size: 1.2em;
  }
  h3 { 
    font-size: 1.1em; 
  }
  h4 { 
    font-size: 1.1em; 
  }
  h5 { 
    font-size: 1em; 
  }
  h6 { 
    font-size: 1em; 
  }
  
  table {
    font-size: 1em;
    padding: 0;
  }
}
```

Example Output:
- [PDF converted with the previous version](https://github.com/travs/markdown-pdf/files/1310608/earlier_version.pdf)
- [PDF converted with the current version](https://github.com/travs/markdown-pdf/files/1310607/current_version.pdf)

**Comment by travs** [comment]
This was an error with a callback if you were using legacy mode.

Unfortunately I am only now getting time to address this, but to any others reading this, if you want to override styles from the default/github stylesheet, you need to use the `.markdown-body` selector, so it might look like this (LESS):

```less
.markdown-body {
  h1 {
    font-size: 1.3em;
  }
  h2 {
    font-size: 1.2em;
  }
  h3 {
    font-size: 1.1em;
  }
  h4 {
    font-size: 1.1em;
  }
  h5 {
    font-size: 1em;
  }
  h6 {
    font-size: 1em;
  }
  table {
    font-size: 1em;
    padding: 0;
  }
}
```

Fixed in the new version (`2.1.3`)

**Comment by holic-cl** [comment]
Hi, i was with same bug, but i didn't have any styles.less file on my ~/.atom folder so i created one with the content posted by @travs
